SEED-platform/seed
# refactor(bsync): use tasks for importing

#### Any background context you want to provide?
Previously BSync files were imported at their own endpoint and parsed in-process (ie blocking).
#### What's this PR do?
Moves the BSync backend import flow to use the flow of other files (ie background tasks). Additionally it adds a mapping page for buildingsync files, though it doesn't currently do anything other than show the mapping from column name to an xpath.
#### How should this be manually tested?
Go through this flow with these different files from the directory `seed/building_sync/tests/data`:
- a supported BuildingSync xml file (v2.0-pr1): `ex_1.xml`
- a zip of supported BSync files: `valid_xml_ex1_ex2.zip`
- a broken BSync file (e.g. missing address info): `ex_1_no_street_address.xml`
1. Navigate to the upload data modal, and click 'Upload BuildingSync Data'.
2. Select the file
3. Click 'continue to mapping', and you should be directed to a mapping page
4. It should show the set of fields from BuildingSync.BRICR_STRUCT and their XPath. Nothing should be editable on this page! Users are not allowed to change mappings currently
5. After clicking "May Your Data", you should be redirected to the mapped properties page. If there was an invalid BSync file included, you should see a modal describing why it failed. If all files failed to be mapped (e.g. a zip with all invalid files, or a single invalid xml file), then you should _not_ see the "save mapping" button, nor should you see any table (it should just say "no data").
6. Go to your properties and the BuildingSync properties should be there.
#### What are the relevant tickets?
#2137
